### PR TITLE
Enh: Improve smart replies prompt

### PR DESCRIPTION
**Pull request by hamza221**

### CODE REVIEW

File: lib/Service/AiIntegrations/AiIntegrationsService.php [hunk 1]
```diff
@@ -196,12 +196,25 @@ public function getSmartReply(Account $account, Mailbox $mailbox, Message $messa
...
+			Please, output *ONLY* a valid JSON string with the keys 'reply1' and 'reply2' for the reply suggestions. Leave out any other text besides the JSON! Be extremely succinct and write the replies from my point of view.
+			 ";
 			$task = new Task(FreePromptTaskType::class, $prompt, 'mail,', $currentUserId);
 			$manager->runTask($task);
-			$replies = array_slice(explode("|", $task->getOutput()), 0, 2);
-			$this->cache->addValue('smartReplies_'.$message->getUid(), implode("|", $replies));
-			return $replies;
+			$replies = $task->getOutput();
+			$this->cache->addValue('smartReplies_'.$message->getId(), $replies);
+			return json_decode($replies, true);
```
Comment (ChristophWurst) [inline lib/Service/AiIntegrations/AiIntegrationsService.php:217]: open the code with phpstorm and let it fix the json_decode to throw on error ;-)

Comment (ChristophWurst) [reply lib/Service/AiIntegrations/AiIntegrationsService.php:217]: and don't cache if you can't decode the replies. else the same decoding error happens again and again.

**Review by ChristophWurst — COMMENTED** [review]
Makes sense

File: lib/Controller/MessagesController.php [hunk 2]
```diff
@@ -908,7 +908,7 @@ public function smartReply(int $messageId):JSONResponse {
 			return new JSONResponse([], Http::STATUS_FORBIDDEN);
 		}
 		try {
-			$replies = $this->aiIntegrationService->getSmartReply($account, $mailbox, $message, $this->currentUserId);
+			$replies = array_values($this->aiIntegrationService->getSmartReply($account, $mailbox, $message, $this->currentUserId));
```
Comment (st3iny) [inline lib/Controller/MessagesController.php:911]: Breaks if the method returns null when the reply of the LLM could not be parsed. Need to check if the result of `getSmartReply()` is null before passing it to `array_values()`.

`array_values(): Argument #1 ($array) must be of type array, null given in file '/home/richard/src/nextcloud/master/dev_apps/mail/lib/Controller/MessagesController.php' line 911`

**Review by st3iny — CHANGES_REQUESTED** [review]

**Comment by st3iny** [comment]
I'm sorry for the late review :see_no_evil:

**Review by st3iny — COMMENTED** [review]
Tested and works.

**Review by st3iny — APPROVED** [review]

**Pull request merged**